com.unity.testing.graphics-performance/Editor/Common/EditorLogWatcher.cs

```
@@ -0,0 +1,53 @@
+using System;
+using System.IO;
+using UnityEngine;
+
+public class EditorLogWatcher : IDisposable
+{
+    const string editorLogFileName = "Editor.log";
+
+    public delegate void OnLogWriteCallback(string newLines);
+
+    OnLogWriteCallback  logWriteCallback;
+    FileStream          logStream;
+
+    public EditorLogWatcher(OnLogWriteCallback callback)
+    {
+        logWriteCallback = callback;
```

You should throw an exception if the callback is `null`.

```cs
logWriteCallback = callback ?? throw new ArgumentNullException(nameof(callback));
```

It does not make sense to use this class with a null callback.